src/git/utils.ts

```diff
@@ -155,14 +158,16 @@ async function listObjects({
   gitDir: string;
   wants: ObjectIdList;
   haves: ObjectIdList;
-}): Promise<ObjectIdList> {
+}/*, ctx: ContextCancellable*/): Promise<ObjectIdList> {
+  // TODO: add support for ctx
   const commits = new Set<string>();
   const trees = new Set<string>();
   const blobs = new Set<string>();
   const tags = new Set<string>();
   const havesSet: Set<string> = new Set(haves);
 
   async function walk(objectId: ObjectId, type: ObjectType): Promise<void> {
+    // ctx.signal.throwIfAborted();
     // If object was listed as a have then we don't need to walk over it
     if (havesSet.has(objectId)) return;
     switch (type) {
```

Comment:
This method is used in `VaultInternal` in `garbageCollectGitObjects`, which is called when initialising `VaultInternal`. As such, we don't have access to a context. Should I manually create a context and pass it in, or make context an optional parameter here?

Reply:
You'll need to pass the `ctx` to whatever needs it. For public methods in a class you should use the `@timedCancellable` decorator. Look at other cancellable methods for reference. For internal methods you can just pass along the `ctx`.

Reply:
After looking through the code more, I have realised that using the `@cancellable` decorator is better for my use case, as it only handles cancellation, while `@timedCancellable` handles both timeouts and cancellation.

The context is only using the signals to signal an abort, so the timeout feature is not needed in most cases.

Reply:
On second thought, perhaps an automatic abort when timeout is reached is actually better than a manual cancellation. I need to have a think about how this works a bit more.